### PR TITLE
sensor: lsm6dsl: cleanup and preparation for migration to DTS props

### PR DESCRIPTION
**Pull request by mniestroj**
Use the same wording for both I2C and SPI variants to improve consistency.

Create `st,lsm6dsl-common.yaml` to include common binding properties from I2C and SPI variants. This
will be used later on to specify new properties just in a single file.

Drop unused/dead code.

Fix minor issues with ODR and range values.

### CODE REVIEW

File: dts/bindings/sensor/st,lsm6dsl-i2c.yaml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Copyright (c) 2017, Linaro Limited
 # SPDX-License-Identifier: Apache-2.0
 
-description: STMicroelectronics LSM6DSL 6-axis accelerometer and gyrometer
+description: |
+  STMicroelectronics LSM6DSL 6-axis accelerometer and gyrometer accessed through
+  I2C bus
```
Comment (avisconti) [inline dts/bindings/sensor/st,lsm6dsl-i2c.yaml:7]: I think you can squash this commit with the dts common file

File: drivers/sensor/lsm6dsl/lsm6dsl.h [hunk 2]
```diff
@@ -667,9 +667,6 @@ struct lsm6dsl_data {
 #endif
 	const struct lsm6dsl_transfer_function *hw_tf;
 	uint16_t accel_freq;
-	uint8_t accel_fs;
-	uint16_t gyro_freq;
-	uint8_t gyro_fs;
```
Comment (avisconti) [inline drivers/sensor/lsm6dsl/lsm6dsl.h:672]: squash this commit with the one with accel_freq

**Review by avisconti — CHANGES_REQUESTED** [review]
LGTM, but I would create the patchset with what indicated here.

File: drivers/sensor/lsm6dsl/lsm6dsl.h (hunk 2 above)
Comment (mniestroj) [reply drivers/sensor/lsm6dsl/lsm6dsl.h:672]: Well, those are two separate logical changes:
- remove unused `accel_fs` and `gyro_{fs,freq}`
- drop double assign of `accel_freq`

and there are reasons provided for each commit. This also makes it much easier to fit into commit title character limits.

Comment (avisconti) [reply drivers/sensor/lsm6dsl/lsm6dsl.h:672]: ok, agreed

**Review by avisconti — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**